app/controllers/idv/hybrid_handoff_controller.rb

```diff
@@ -0,0 +1,230 @@
+module Idv
+  class HybridHandoffController < ApplicationController
+    include ActionView::Helpers::DateHelper
+    include IdvSession
+    include IdvStepConcern
+    include StepIndicatorConcern
+    include StepUtilitiesConcern
+
+    before_action :confirm_two_factor_authenticated
+    before_action :render_404_if_hybrid_handoff_controller_disabled
+    before_action :confirm_agreement_step_complete
+
+    def show
+      flow_session[:flow_path] = 'standard'
+      analytics.idv_doc_auth_upload_visited(**analytics_arguments)
+
+      Funnel::DocAuth::RegisterStep.new(current_user.id, sp_session[:issuer]).call(
+        'upload', :view,
+        true
+      )
+
+      render :show, locals: extra_view_variables
+    end
+
+    def update
+      irs_attempts_api_tracker.idv_document_upload_method_selected(
+        upload_method: params[:type],
+      )
+
+      # See the simple_form_for in
+      # app/views/idv/doc_auth/upload.html.erb
+      if hybrid_flow_chosen?
+        handle_phone_submission
+      else
+        bypass_send_link_steps
+      end
+    end
+
+    def hybrid_flow_chosen?
+      params[:type] != 'desktop' && !mobile_device?
+    end
+
+    def handle_phone_submission
+      throttle.increment!
+      return throttled_failure if throttle.throttled?
+      idv_session.phone_for_mobile_flow = params[:doc_auth][:phone]
+      flow_session[:phone_for_mobile_flow] = idv_session.phone_for_mobile_flow
+      telephony_result = send_link
+      telephony_form_response = build_telephony_form_response(telephony_result)
+
+      failure_reason = nil
+      if !telephony_result.success?
+        failure_reason = { telephony: [telephony_result.error.class.name.demodulize] }
+        failure(telephony_form_response.errors[:message])
+      end
+      irs_attempts_api_tracker.idv_phone_upload_link_sent(
+        success: telephony_result.success?,
+        phone_number: formatted_destination_phone,
+        failure_reason: failure_reason,
+      )
+
+      if !failure_reason
+        flow_session[:flow_path] = 'hybrid'
+        redirect_to idv_link_sent_url
+      else
+        redirect_to idv_hybrid_handoff_url
+      end
+
+      analytics.idv_doc_auth_upload_submitted(
+        **analytics_arguments.merge(form_response(destination: :link_sent).to_h),
+      )
+
+      telephony_form_response
```

Comment:
Returning the form response object is a relic of the flow state machine. I don't think we need it anymore. I think we should move the analytics call up with the `irs_attempts_api_tracker` call and then let the `failure_reason` conditional be the last thing we do here.

Reply:
Unfortunately `analytics_arguments` needs to read from the flow_session to get the correct `:flow_path`, and that is only set on line 62 in hybrid cases. If analytics is moved before that line it will report the incorrect flow path. But agree on returning the response

Reply:
gotcha. regardless I don't think we need this return